drivers/unix/os_unix.cpp

```diff
@@ -199,10 +199,7 @@ static int has_data(FILE *p_fd, int timeout_usec = 0) {
 	int fd = fileno(p_fd);
 	FD_ZERO(&readset);
 	FD_SET(fd, &readset);
-	timeval time;
-	time.tv_sec = 0;
-	time.tv_usec = timeout_usec;
-	int res = 0; //select(fd + 1, &readset, NULL, NULL, &time);
+	int res = 0;
```

Comment:
Actually this part is still weird. The method returns an `int`, and given how it's written it will always be `false`, so `0`.
`has_data` is then only used in https://github.com/hpvb/godot/blob/6bf377e0e307adaa0f4a8fa30d416b1ccde8b36f/drivers/unix/os_unix.cpp#L216-L228, so this code chunk is never executed since the method always returns false/0.

So I was wrong to tell you to remove that code, I think it would be better (for this commit) to keep it with a `// FIXME` comment, and open an issue so that it can be investigated and fixed.

Reply:
I'll see about actually fixing this function then.